src/Trakx.CoinGecko.ApiClient/Configuration/CoinGeckoApiConfiguration.cs

```diff
@@ -4,27 +4,31 @@
 
 namespace Trakx.CoinGecko.ApiClient;
 
+/// <summary>Main configuration for how to connect to the CoinGecko API.</summary>
 public record CoinGeckoApiConfiguration
 {
+    /// <summary>Base Url for all calls to the API. Default is https://pro-api.coingecko.com/api/v3/</summary>
+    [AwsParameter(AllowGlobal = true)]
     public Uri BaseUrl { get; init; } = Constants.ProBaseUrl;
 
-    public int MaxRetryCount { get; init; } = 10;
-
-    public TimeSpan InitialRetryDelay { get; init; } = TimeSpan.FromMilliseconds(100);
-
-    // this setting isn't used anywhere
-    public int? ThrottleDelayPerSecond { get; init; } = 120;
+    /// <summary>Access key added to all requests to the API.</summary>
+    [AwsParameter(AllowGlobal = true)]
+    [SecretEnvironmentVariable]
+    public string ApiKey { get; init; } = string.Empty;
 
+    /// <summary>How long to cache requests to the API, to reuse them and reduce usage. Default is 10 seconds.</summary>
+    [AwsParameter(AllowGlobal = true)]
```

Comment:
I'm not sure if the `CacheDuration` should be an AWS Parameter. 🤔 
If we decide to override this cache duration, we will need to create a specific AWS parameter in our service to override, being not able to override via `appsettings.json`.
Is that the expected behaviour? If yes, I'm okay with it, just confirming 😉

Reply:
Sorry, I don't understand. I thought allowing something to be an AWS parameter just means we could set a value for it and restart the service and that's it, no need for PRs?

Basically to prevent what I had to do here:
https://github.com/trakx/market-data-snapshots/pull/73

I think what your "fear" is that this could be set as a global parameter, yes?
In that case, I will use Matt's defense of: we are all responsible adults, we won't do that.

Reply:
I can remove the "AllowGlobal" on it, it would prevent the scenario you described, right ?

Reply:
I just think that there's no point in reducing features just because we might do something wrong... that's on us to handle responsibily, not the tech to limit us.

Reply:
CacheDuration is no longer allowed to be an AWS parameter.